x/merkledb/proof.go

```diff
@@ -190,6 +194,53 @@ func (proof *Proof) Verify(ctx context.Context, expectedRootID ids.ID) error {
 	return nil
 }
 
+func (proof *Proof) ToProto() *pb.Proof {
+	value := &pb.MaybeBytes{}
+	if !proof.Value.IsNothing() {
+		value.Value = proof.Value.Value()
+	} else {
+		value.IsNothing = true
+	}
+
+	pbProof := &pb.Proof{
+		Key:   proof.Key,
+		Value: value,
+	}
+
+	pbProof.Proof = make([]*pb.ProofNode, len(proof.Path))
+	for i, node := range proof.Path {
+		pbProof.Proof[i] = node.ToProto()
+	}
+
+	return pbProof
+}
+
+func (proof *Proof) UnmarshalProto(pbProof *pb.Proof) error {
+	switch {
+	case pbProof == nil:
+		return ErrNilProof
+	case pbProof.Value == nil:
```

Comment:
Isn't this valid?

Reply:
This type is a `*pb.MaybeBytes` so if the value is nil we expect it to be `MaybeBytes{Value: nil, IsNothing: false}`, not `nil`

Reply:
ahh, got it.